lib/health-indicator/microservice/external/serializer.interface.ts

```diff
@@ -0,0 +1,11 @@
+import { OutgoingEvent, OutgoingResponse, OutgoingRequest } from './packet.interface';
+
+export interface Serializer<TInput = any, TOutput = any> {
+  serialize(value: TInput): TOutput;
+}
```

Comment:
Any reason why it has to be duplicated?

Reply:
The types of `@nestjs/microservices` always ended up in my definition files, as soon as I included it in the public API of Terminus. Therefore it tried to load the types of it and then TypeScript complains. You could disable this warning by having `skipLibCheck` in your tsconfig.json, but I did not want that projects need to have this option enabled in order to use Terminus.

Reply:
I've faced the same issue with the `@nestjs/core` and `@nestjs/microservices`. The `@nestjs/microservices` package is optional and is being lazily loaded by the core package (on demand). However, the `NestFactory#createMicroservice` method takes the `MicroserviceOptions` which lives in the lazy loaded package (microservices). To fix this issue, I changed it to a generic https://github.com/nestjs/nest/blob/master/packages/core/nest-factory.ts#L93

Likewise, in your case you need all these redundant interfaces for this type: https://github.com/nestjs/terminus/pull/593/files#diff-de039754bac7c557b30163c56d080e57R16 (I guess?).  You could change it to a generic:

```typescript
export type MicroserviceHealthIndicatorOptions<T extends object> = T & {
  timeout?: number;
};
```

And obviously, to get code autocompletion, developers will have to pass it as a type argument `<MicroserviceOptions>`, but I suppose it's much better than having all interfaces deduplicated :D 

If you think `object` isn't sufficient constraint, you can use sth like `Record<'transport' | 'options', any>` instead

Reply:
Smart! I think, for the time being, this suits me. Though I may take the mono repo solution in the future, so I do not have to worry about these things anymore

Reply:
Fixed